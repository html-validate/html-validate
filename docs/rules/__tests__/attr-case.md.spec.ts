import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<p ID="foo"></p>`;
markup["correct"] = `<p id="foo"></p>`;
markup["svg-viewbox"] = `<svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" />`;

describe("docs/rules/attr-case.md", () => {
	it("inline validation: incorrect", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"attr-case":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"attr-case":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: svg-viewbox", () => {
		const htmlvalidate = new HtmlValidate({"rules":{"attr-case":"error"}});
		const report = htmlvalidate.validateString(markup["svg-viewbox"]);
		expect(report.results).toMatchSnapshot();
	});
});
