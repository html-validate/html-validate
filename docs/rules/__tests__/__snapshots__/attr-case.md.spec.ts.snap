// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/attr-case.md inline validation: correct 1`] = `Array []`;

exports[`docs/rules/attr-case.md inline validation: incorrect 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 4,
        "context": undefined,
        "line": 1,
        "message": "Attribute \\"ID\\" should be lowercase",
        "offset": 3,
        "ruleId": "attr-case",
        "severity": 2,
        "size": 2,
      },
    ],
    "source": "<p ID=\\"foo\\"></p>",
    "warningCount": 0,
  },
]
`;

exports[`docs/rules/attr-case.md inline validation: multiple 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 4,
        "context": undefined,
        "line": 3,
        "message": "Attribute \\"fooBar\\" should be lowercase or uppercase",
        "offset": 33,
        "ruleId": "attr-case",
        "severity": 2,
        "size": 6,
      },
    ],
    "source": "<p foobar></p>
<p FOOBAR></p>
<p fooBar></p>",
    "warningCount": 0,
  },
]
`;

exports[`docs/rules/attr-case.md inline validation: svg-viewbox 1`] = `Array []`;
